networkx/algorithms/approximation/steinertree.py

```diff
@@ -86,15 +256,37 @@ def steiner_tree(G, terminal_nodes, weight="weight"):
     References
     ----------
     .. [1] Steiner_tree_problem on Wikipedia.
-       https://en.wikipedia.org/wiki/Steiner_tree_problem
+           https://en.wikipedia.org/wiki/Steiner_tree_problem
+    .. [2] Kou, L., G. Markowsky, and L. Berman. 1981.
+           ‘A Fast Algorithm for Steiner Trees’.
+           Acta Informatica 15 (2): 141–45.
+           https://doi.org/10.1007/BF00288961.
+    .. [3] Wu, Y. F., P. Widmayer, and C. K. Wong. 1986.
+           ‘A Faster Approximation Algorithm for the Steiner Problem in Graphs’.
+           Acta Informatica 23 (2): 223–29.
+           https://doi.org/10.1007/BF00289500.
+    .. [4] Mehlhorn, Kurt. 1988.
+           ‘A Faster Approximation Algorithm for the Steiner Problem in Graphs’.
+           Information Processing Letters 27 (3): 125–28.
+           https://doi.org/10.1016/0020-0190(88)90066-X.
     """
-    # H is the subgraph induced by terminal_nodes in the metric closure M of G.
-    M = metric_closure(G, weight=weight)
-    H = M.subgraph(terminal_nodes)
-    # Use the 'distance' attribute of each edge provided by M.
-    mst_edges = nx.minimum_spanning_edges(H, weight="distance", data=True)
-    # Create an iterator over each edge in each shortest path; repeats are okay
-    edges = chain.from_iterable(pairwise(d["path"]) for u, v, d in mst_edges)
+    if method is None:
+        import warnings
+
+        msg = (
+            "steiner_tree will change default method from 'kou' to 'wu' "
+            "in version 3.0.\nSet the `method` kwarg to remove this warning."
```

Comment:
I think this deprecation will need to get pushed out to v3.2 because we've taken so long to review this PR.